openwisp_notifications/api/permissions.py

```diff
@@ -0,0 +1,17 @@
+from rest_framework.permissions import BasePermission
+
+
+class IsAuthenticatedToUpdateNotificationSetting(BasePermission):
+    def has_permission(self, request, view):
+        user = request.user
+        user_id = view.kwargs.get('user_id')
+
+        # Allow superusers
+        if user.is_superuser:
+            return True
+
+        # Allow users to update their own settings
+        if user.id == user_id:
+            return True
+
+        return False
```

Comment:
This code can be condensed as 

```suggestion
        return request.user.is_superuser or request.user.id == view.kwargs.get('user_id')
```

Please add a docstring explaining that only superuser can change the notification preferences of other users.